iceoryx_posh/source/roudi/port_manager.cpp

```diff
@@ -245,27 +244,30 @@ void PortManager::handleInterfaces()
 
 void PortManager::handleApplications()
 {
-    capro::CaproMessage l_caproMessage;
+    capro::CaproMessage caproMessage;
 
-    for (auto l_applicationPortData : m_portPool->appliactionPortDataList())
+    for (auto applicationPortData : m_portPool->appliactionPortDataList())
     {
-        iox::popo::ApplicationPort l_applicationPort(l_applicationPortData);
-        while (l_applicationPort.getCaProMessage(l_caproMessage))
+        iox::popo::ApplicationPort applicationPort(applicationPortData);
+
+        for (auto maybeCaproMessage = applicationPort.getCaProMessage(); maybeCaproMessage.has_value();
+             maybeCaproMessage = applicationPort.getCaProMessage())
```

Comment:
😎

Reply:
reminds me that I wanted try to code something like this in C++ for ages
```
for(auto caproMessage: applicationPort.iterCaproMessage())
{
    // do something with caproMessage
}
```
something generic which calls getNext until it's a `nullopt`

Reply:
It just ocured to me that we already can do something similar
```
while (auto caproMessage = applicationPort.getCaProMessage())
{
    // do something with capro message
}
```
the while loop uses the bool operator of the optional and stops iterating if it's a nullopt instead of a caproMessage